cpp/benchmarks/common/generate_input.hpp

```diff
@@ -399,6 +399,88 @@ class data_profile {
   }
 };
 
+/**
+ * @brief Builder to build data profile for the random data generator.
+ */
+class data_profile_builder {
```

Comment:
This could be for a future PR, but if we want this to be a standard mechanism then I think it is worth some additional documentation and examples both here and in the dev docs: https://github.com/rapidsai/cudf/blob/branch-22.10/cpp/docs/BENCHMARKING.md

Reply:
I'll add some docs here in this PR, but [BENCHMARKING.md](https://github.com/rapidsai/cudf/blob/branch-22.10/cpp/docs/BENCHMARKING.md) might be better as a separate PR, since large changes are needed - there's nothing about `data_profile` there ATM.